hax/hax/queue/__init__.py

```diff
@@ -96,3 +99,17 @@ def to_ha_state(self, objinfo: Dict[str, str]) -> Optional[HAState]:
             LOG.error('Invalid json payload, no key (%s) present', error)
             return None
         return HAState(sdev_fid, status=state)
+
+    def handle_ha_notify(self, payload: List[Dict[str, Any]]) -> None:
+        hastates: List[HAState] = []
+        for state in payload:
+            hastatus = ServiceHealth.UNKNOWN
+            if state['status'] == 'FAILED':
+                hastatus = ServiceHealth.FAILED
+            elif state['status'] == 'OK':
+                hastatus = ServiceHealth.OK
+            hastates.append(HAState(Fid.parse(state['fid']), hastatus))
+
+        if hastates:
+            self.queue.put(BroadcastHAStates(states=hastates,
+                                             reply_to=None))
```

Comment:
This call will not wait for delivery. Can't it cause problems in the future?

I guess we had to introduce DeliveryHerald specially for queue processing semantics: we don't want to mark a message in the queue as read until we can't be sure that the message is delivered to Motr world for sure. Isn't it the same case?


If so, this piece of code could look like this (draft, didn't check for correctness):
```python
        q: Queue = Queue(1)
        self.queue.put(
            BroadcastHAStates(BroadcastHAStates(states=hastates,
                                                reply_to=q))
        ids: List[MessageId] = q.get()
        self.herald.wait_for_any(HaLinkMessagePromise(ids))
```

Reply:
Yeah, I did had waiting part earlier but removed to see how it affects. It worked but let's add waiting here too to maintain the semantics.

Reply:
Reverted these changes.